app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -2,10 +2,34 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   include PreferencesController
 
   require_authentication :all, scopes: [:project]
+  extra_schema_actions :update_settings
   schema_type :json_schema
+  before_action :require_project_ownership, only: [:update_project_settings]
+
+
```

Comment:
as for the failing spec - try and see if you get to the before_action filter or not..if you do then trace it down...if not it might be a thing with controlled resources...